Remove stdenv adapter cruft

Artturin opened the pull request:
See commit msgs
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Ericson2314 commented:
I would post to Nixpkgs to make sure no one is using or. Or maybe @siraben can teach us how to do a deprecation cycle like `stdenv.lib` -> `lib` used?

Artturin commented:
> I would post to Nixpkgs to make sure no one is using or. Or maybe @siraben can teach us how to do a deprecation cycle like `stdenv.lib` -> `lib` used?

i searched github for them but all i can find is nixpkgs forks

eg. `useGoldLinker` is still being used

Ericson2314 commented:
Well, on principle, we should still do deprecation cycles. It shouldn't be too hard, and there is no good thing that this stuff is blocking.

As one that loves cleaning things up and thinks stability is overrated in many quarters, I've nonetheless learned that deprecation cycles are good politics and simply polite. If 9 times out of ten no one is using the thing, but 1 time out of 10 you hear back from a user you missed searching, it is worth it.

Artturin committed:
stdenv: deprecate unused adapters

stdenv: deprecate addCoverageInstrumentation adapter

this used to be used in nixos/tests but it hasn't been used in nixpkgs
for years

stdenv: deprecate replaceMaintainersField adapter

it was added in 2009 in 01e98e49b155d989d8c80066db6de910396cdcaa
by nbp
there are no uses of it in nixpkgs now

stdenv: deprecate validateLicenses adapter

it was added in 2009 in b29073af25cb0ff2d8d9a0aa1f958316904d3dc9

unfreePredicate is now handled in ./pkgs/stdenv/generic/check-meta.nix
so this is unnecessary

siraben reviewed (approved):
Looks good to me, the trace is what we did when we deprecated `stdenv.lib` and that eased migration since we got pings from different projects that saw the message.

Ericson2314 commented:
@siraben how was that trace done?

Ericson2314 commented:
@siraben In particular, wasn't there a way you use `config` or something to CI nixpkgs with the deprecated items removed to ensure usages did not crop back in?

Sorry @Artturin I just want to understand the full process so we can replecate it this time and many others.

Artturin commented:
https://github.com/NixOS/nixpkgs/pull/111284

Ericson2314 commented:
Huh I thought there was more but I won't belabor it :).
